### PR TITLE
fixing battery display

### PR DESCRIPTION
**Pull request by Fro99666**
- removing extra % on display / removing useless extra code / manage
"energy_" + "charge_"  case

### CODE REVIEW

File: lib/class.OS_Linux.php [hunk 1]
```diff
@@ -935,18 +935,19 @@ public function getBattery() {
...
-			$go_for_it = true;
-
-			foreach(array($b.'/manufacturer', $b.'/status', $b.'/charge_now') as $f)
-				if (!is_file($f))
-					$go_for_it = false; // Continue out of two nested loops
-
-			if (!$go_for_it) 
-				continue;
+			foreach(array($b.'/manufacturer', $b.'/status') as $f)
+				if (!is_file($f))continue;
```
Comment (jrgp) [inline lib/class.OS_Linux.php]: The point of this is to break out of the $bats loop if either of those files are missing. The original code satisfies this; to do it with smaller code, you'll need to do "continue 2;" instead.

Also don't put the continue on the same line as the if; normal indention applies.

File: lib/class.OS_Linux.php [hunk 2]
```diff
@@ -935,18 +935,19 @@ public function getBattery() {
...
-			$charge_now = LinfoCommon::getIntFromFile($b.'/charge_now');
+			if(is_file($b.'/charge_full'))
+				{
+				$charge_full = LinfoCommon::getIntFromFile($b.'/charge_full');
+				$charge_now = LinfoCommon::getIntFromFile($b.'/charge_now');
+				}
+			else{
+				$charge_full = LinfoCommon::getIntFromFile($b.'/energy_full');
+				$charge_now = LinfoCommon::getIntFromFile($b.'/energy_now');
+				}
```
Comment (jrgp) [inline lib/class.OS_Linux.php]: Please mind the braces and indention. Keep a space between the else and { and keep the indention levels consistent.

File: lib/class.OS_Linux.php [hunk 3]
```diff
@@ -955,7 +956,7 @@ public function getBattery() {
 			$return[] = array(
 				'charge_full' => $charge_full,
 				'charge_now' => $charge_now,
-				'percentage' => (is_numeric($percentage) && $percentage > 100 ? 100 : $percentage ).'%',
+				'percentage' => (is_numeric($percentage) && $percentage > 100 ? 100 : $percentage ),
```
Comment (jrgp) [inline lib/class.OS_Linux.php]: Remove space between e and )

**Comment by Fro99666** [comment]
sorry i didn't pay attention !
i am used to { and } on same line 
thanks for the "continue 2" i didn't known this notation

File: lib/class.OS_Linux.php [hunk 4]
```diff
@@ -935,18 +935,23 @@ public function getBattery() {
...
+				case is_file($b.'/energy_full'):
+					$charge_full = LinfoCommon::getIntFromFile($b.'/energy_full');
+					$charge_now = LinfoCommon::getIntFromFile($b.'/energy_now');
+					break;
+				case is_file($b.'/charge_full'):
+					$charge_full = LinfoCommon::getIntFromFile($b.'/charge_full');
+					$charge_now = LinfoCommon::getIntFromFile($b.'/charge_now');
+					break;
+				default:
+					continue;
```
Comment (jrgp) [inline lib/class.OS_Linux.php]: One last thing, promise! Can you please put a break after this continue?

It isn't strictly needed but it does aid readability and will reduce the chances of bad things happening later if more code is added to the switch's body.

Comment (Fro99666) [inline lib/class.OS_Linux.php]: it is ok if i do it monday ?  i am not on my comp with the repo and i ll be busy this week end.
By the way dont worry about your request, i prefere to make it as you want, as it is your code, i won't mess it !

**Comment by jrgp** [comment]
Sure, take your time. :)

**Comment by Fro99666** [comment]
ok syntaxe as been uniformized ^_^

i'll be in holidays for 2 weeks starting next friday, i am going to visit Minesota !
so i will not able to pull any request during this period.

If i see anything else, i ll will do it when i'll be back.

Tchuss !

**Comment by Fro99666** [comment]
my windows 2012 doesn't send cpu info, so i added a fix, result can be seen here:
http://info.test.frogg.fr/ (win2012)
http://info.pt.frogg.fr/ (win2008)

File: lib/class.OS_Windows.php [hunk 5]
```diff
@@ -182,10 +182,13 @@ public function getCPU() {
 			$curr = array(
 				'Model' => $cpu->Name,
 				'Vendor' => $cpu->Manufacturer,
-				'MHz' => $cpu->CurrentClockSpeed,
-				'usage_percentage' => $cpu->LoadPercentage,
+				'MHz' => $cpu->CurrentClockSpeed
 			);
 			
+			if($cpu->LoadPercentage!=''){
```
Comment (jrgp) [inline lib/class.OS_Windows.php]: Please spaces around != operators

File: lib/class.OS.php [hunk 6]
```diff
@@ -28,7 +28,7 @@ public function __call($name, $args) {
 	}
 
 	public function getAccessedIP() {
-		return isset($_SERVER['SERVER_ADDR']) && $_SERVER['SERVER_ADDR'] ? $_SERVER['SERVER_ADDR'] : 'Unknown';
+		return isset($_SERVER['SERVER_ADDR']) && $_SERVER['SERVER_ADDR'] ? $_SERVER['SERVER_ADDR'] : isset($_SERVER['LOCAL_ADDR']) && $_SERVER['LOCAL_ADDR'] ? $_SERVER['LOCAL_ADDR'] : 'Unknown';
```
Comment (jrgp) [inline lib/class.OS.php]: Please put nested ternaries in parens

**Comment by Fro99666** [comment]
i am not sure about what mean "Please put nested ternaries in parens"
so i added () to the 2nd condition, i hope it is what you meant !

File: lib/class.OS_Windows.php [hunk 7]
```diff
@@ -182,10 +182,13 @@ public function getCPU() {
 			$curr = array(
 				'Model' => $cpu->Name,
 				'Vendor' => $cpu->Manufacturer,
-				'MHz' => $cpu->CurrentClockSpeed,
-				'usage_percentage' => $cpu->LoadPercentage,
+				'MHz' => $cpu->CurrentClockSpeed
 			);
 			
+			if($cpu->LoadPercentage != ''){
```
Comment (jrgp) [inline lib/class.OS_Windows.php]: Really sorry but last thing I promise! Please put a space between the ( and {

**Comment by jrgp** [comment]
Yep that's what I meant. I just wrote one last comment and then I'll merge when it's fixed.

**Comment by Fro99666** [comment]
ok, done ^_^

**Comment by jrgp** [comment]
Thanks!

**Pull request merged**